grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/dirty/GrailsEntityDirtinessStrategy.groovy

```diff
@@ -56,6 +60,30 @@ class GrailsEntityDirtinessStrategy implements CustomEntityDirtinessStrategy {
     void resetDirty(Object entity, EntityPersister persister, Session session) {
         if (canDirtyCheck(entity, persister, session)) {
             cast(entity).trackChanges()
+            try {
+                PersistentEntity persistentEntity = GormEnhancer.findEntity(Hibernate.getClass(entity))
+                if (persistentEntity != null) {
+                    resetDirtyEmbeddedObjects(persistentEntity, entity, persister, session)
+                }
+            } catch (IllegalStateException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(e.message, e)
+                }
+            }
+        }
+    }
+
+    private void resetDirtyEmbeddedObjects(PersistentEntity persistentEntity,
+                                           Object entity,
+                                           EntityPersister persister,
+                                           Session session) {
+
+        if (DirtyCheckingSupport.areEmbeddedDirty(persistentEntity, entity)) {
+            final associations = persistentEntity.getEmbedded()
+            for (Embedded a in associations) {
+                final value = a.reader.read(entity)
+                resetDirty(value, persister, session)
```

Comment:
What if `value` is null?

Reply:
If the value is `null` then we do not need to reset `$changedProperties` on embedded. So, the check at line#81 `DirtyCheckingSupport.areEmbeddedDirty(persistentEntity, entity)` would return `false`.